### ISSUE TITLE
Schema for argument_specs.yml doesn't allow complex requirements (mutually_exclusive, required_if and others)

### ISSUE THREAD

**Issue by tapetersen** [ansible/ansible-lint#3822]
##### Summary

The schema for argument_specs.yml doesn't support the more advanced conditions:
 * required_by
 * mutually_exclusive
 * required_together
 * required_one_of
 * required_if

As it doesn't allow additional properties ansible-lint will flag valid files as invalid.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

Ran in current devenv from main

```console
$ ansible-lint --version
ansible-lint 0.1.dev2609 using ansible-core:2.15.4 ansible-compat:4.1.10 ruamel-yaml:0.17.33 ruamel-yaml-clib:0.2.8
A new release of ansible-lint is available: 0.1.dev2609 → 6.20.3 Upgrade by running: pip install --upgrade ansible-lint
```
Found in vscode as part of the ansible plugin from redhat but it comes ultimately from here from what I understand.

##### STEPS TO REPRODUCE

argument_spec.yml as below will have errors.

```yaml
---
# https://docs.ansible.com/ansible/latest/user_guide/playbooks_reuse_roles.html#role-argument-validation
argument_specs:
  main:
    options:
      complex_required_options:
        type: dict
        options:
          foo:
            type: str
          bar:
            type: str
          baz:
            type: str

        mutually_exclusive:
          - ["foo", "bar"]

        required_together:
          - ["bar", "baz"]

        required_one_of:
          - ["foo", "bar", "baz"]

        required_if:
          - ["foo", "must_have_bar_and_baz_default", ["bar", "baz"]]
          - ["foo", "must_have_bar_and_baz_explicit", ["bar", "baz"], false]
          - ["foo", "must_have_one_of_bar_or_baz", ["bar", "baz"], true]

        required_by:
          foo: "bar"
          bar: ["foo", "baz"]

```

##### Desired Behavior

This should at least be accepted and at best be checked for validity according to scheam.

##### Actual Behavior

![image](https://github.com/ansible/ansible-lint/assets/380497/728e48fc-4a34-41e4-842f-fe4a12e3e966)
